opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java

```diff
@@ -0,0 +1,208 @@
+
+        String hint = RuntimeEnvironment.application.getResources().getString(R.string.search_hint);
+        Assert.assertEquals(hint, provider.searchHint());
+    }
+
+    @Test
+    @Ignore
+    public void onCreateView() {
+
+        View parentLayout = LayoutInflater.from(RuntimeEnvironment.application.getApplicationContext()).inflate(R.layout.fragment_base_register, null, false);
+        Mockito.doReturn(parentLayout).when(layoutInflater).inflate(R.layout.fragment_base_register, container, false);
+        Mockito.doReturn(activity).when(baseRegisterFragment).getActivity();
+
+        baseRegisterFragment.onCreateView(layoutInflater, container, bundle);
+    }
+
+    @Test
+    public void assertGetLayoutReturnsCorrectLayout() {
+
+        Assert.assertEquals(R.layout.fragment_base_register, baseRegisterFragment.getLayout());
+    }
+
+    @Test
+    public void assertUpdateSearchViewAddsCorrectListnersToSearchView() {
+
+        Mockito.doReturn(searchView).when(baseRegisterFragment).getSearchView();
+
+        ReflectionHelpers.setField(baseRegisterFragment, "textWatcher", textWatcher);
+        ReflectionHelpers.setField(baseRegisterFragment, "hideKeyboard", hideKeyboard);
+
+        baseRegisterFragment.updateSearchView();
+
+        Mockito.verify(searchView).removeTextChangedListener(textWatcher);
+        Mockito.verify(searchView).addTextChangedListener(textWatcher);
+        Mockito.verify(searchView).setOnKeyListener(hideKeyboard);
+    }
+
+    @Test
+    public void assertUpdateSearchBarHintSetsCorrectValue() {
+
+        Mockito.doReturn(searchView).when(baseRegisterFragment).getSearchView();
+
+        baseRegisterFragment.updateSearchBarHint(TEST_RANDOM_STRING);
+
+        Mockito.verify(searchView).setHint(TEST_RANDOM_STRING);
+    }
+
+    @Test
+    public void setSearchTermInitsCorrectValue() {
+
+        Mockito.doReturn(searchView).when(baseRegisterFragment).getSearchView();
+
+        baseRegisterFragment.setSearchTerm(TEST_RANDOM_STRING);
+
+        Mockito.verify(searchView).setText(TEST_RANDOM_STRING);
+    }
+
+    @Test
+    public void assertOnQRCodeSucessfullyScannedInvokesFilterWithCorrectParams() {
+
+        String OPENSRP_ID = "8232-372-8L";
+        String OPENSRP_ID_NO_HYPHENS = "82323728L";
+
+        baseRegisterFragment = Mockito.spy(baseRegisterFragment);
+
+        Mockito.doReturn(searchCancelView).when(baseRegisterFragment).getSearchCancelView();
+
+        Mockito.doNothing().when(baseRegisterFragment).filter(ArgumentMatchers.eq(OPENSRP_ID_NO_HYPHENS), ArgumentMatchers.eq(""), ArgumentMatchers.anyString(), ArgumentMatchers.eq(true));
+
+        baseRegisterFragment.onQRCodeSucessfullyScanned(OPENSRP_ID);
+
+        Mockito.verify(baseRegisterFragment).filter(openSRPIdArgumentCaptor.capture(), stringArgumentCaptor.capture(), stringArgumentCaptor.capture(), qrCodeArgumentCaptor.capture());
```

Comment:
Verify the boolean passed in the `filter` method as the last param

Reply:
Done